lib/msal-browser/src/interaction_handler/SilentHandler.ts

```diff
@@ -0,0 +1,175 @@
+        const frameName = "msalIdTokenFrame";
+        return this.loadFrameTimeout ? await this.loadFrame(requestUrl, frameName) : this.loadFrameSync(requestUrl, frameName);
+    }
+
+    /**
+     * Monitors an iframe content window until it loads a url with a known hash, or hits a specified timeout.
+     * @param iframeContentWindow 
+     * @param timeout 
+     * @param urlNavigate 
+     */
+    monitorFrameForHash(iframe: HTMLIFrameElement, timeout: number, urlNavigate: string): Promise<string> {
+        const contentWindow = iframe.contentWindow;
+        return new Promise((resolve, reject) => {
+            const maxTicks = timeout / BrowserConstants.POLL_INTERVAL_MS;
+            let ticks = 0;
+
+            const intervalId = setInterval(() => {
+                if (contentWindow.closed) {
+                    // Window is closed
+                    this.removeHiddenIframe(iframe);
+                    clearInterval(intervalId);
+                    reject(BrowserAuthError.createIframeClosedPrematurelyError());
+                    return;
+                }
+
+                let href;
+                try {
+                    /*
+                     * Will throw if cross origin,
+                     * which should be caught and ignored
+                     * since we need the interval to keep running while on STS UI.
+                     */
+                    href = contentWindow.location.href;
+                } catch (e) {}
+
+                /*
+                 * Always run clock for silent calls
+                 * as silent operations should be short,
+                 * and to ensure they always at worst timeout.
+                 */
+                ticks++;
+
+                if (UrlString.hashContainsKnownProperties(href)) {
+                    // Success case
+                    const contentHash = contentWindow.location.hash;
+                    this.removeHiddenIframe(iframe);
+                    clearInterval(intervalId);
+                    resolve(contentHash);
+                    return;
+                } else if (ticks > maxTicks) {
+                    // Timeout error
+                    this.removeHiddenIframe(iframe);
+                    clearInterval(intervalId);
+                    reject(BrowserAuthError.createMonitorWindowTimeoutError(urlNavigate));
+                    return;
+                }
+            }, BrowserConstants.POLL_INTERVAL_MS);
+        });
+    }
+    
+    /**
+     * @hidden
+     * Loads iframe with authorization endpoint URL
+     * @ignore
+     */
+    private loadFrame(urlNavigate: string, frameName: string): Promise<HTMLIFrameElement> {
+        /*
+         * This trick overcomes iframe navigation in IE
+         * IE does not load the page consistently in iframe
+         */
+
+        return new Promise((resolve, reject) => {
+            setTimeout(() => {
+                const frameHandle = this.loadFrameSync(urlNavigate, frameName);
```

Comment:
I thought the idea of `loadFrameSync` is to have the code outside of `setTimeOut()` if the user sets it to `0`

Reply:
That happens on line 25

Reply:
loadFrame just calls loadFrameSync internally